File: lib/google_cloud/pub_sub.rb
require 'google/cloud/pubsub'

module GoogleCloud
  class PubSub
    def initialize(topic)
      @topic = self.class.client.topic topic
    end

    def publish(message)
      @topic.publish message.to_json
    end

    def self.client
      @client ||= ::Google::Cloud::Pubsub.new project: ENV['GC_PROJECT_ID']
    end
  end
end

File: ml/vcwiz/server/pubsub.py
import os, json
from google.cloud import pubsub_v1

def subscribe(subscription, callback):
  subscriber = pubsub_v1.SubscriberClient()
  path = subscriber.subscription_path(os.environ['GC_PROJECT_ID'], subscription)
  subscriber.subscribe(path, callback=callback)

def push(topic, message):
  publisher = pubsub_v1.PublisherClient()
  path = publisher.topic_path(os.environ['GC_PROJECT_ID'], topic)
  publisher.publish(path, message)
